File: problem-001/problem1.py
# Project Euler
# Problem 1

multiplesCounter = 0
multiplesSum = 0

def multiples(n):
    global multiplesCounter
    global multiplesSum
    for x in range(0,n):
        if multipleOf(x,3) or multipleOf(x,5) :
            multiplesCounter += 1
            multiplesSum += x

def multipleOf(n,x):
    if n % x == 0:
        return True
    else:
        return False

multiples(1000)
print multiplesSum

File: problem-001/problem1.js
// Project Euler
// Problem 1

function* getMultiplesOf(upperBound, terms) {
  let current = 1;
  while (current < upperBound) {
    if (terms.some((term) => current % term === 0)) {
      yield current;
    }
    current++;
  }
}

const multiples = getMultiplesOf(1000, [3, 5]);

let sumOfMultiples = 0;
let currentMultiple = multiples.next().value;

while (currentMultiple) {
  sumOfMultiples += currentMultiple;
  currentMultiple = multiples.next().value;
}

console.log(sumOfMultiples);